hs_core/views/__init__.py

```diff
@@ -245,9 +245,100 @@ def delete_file(request, shortkey, f, *args, **kwargs):
 def delete_resource(request, shortkey, *args, **kwargs):
     res, _, _ = authorize(request, shortkey, edit=True, full=True, superuser=True)
 
+    if res.metadata.relations.all().filter(type='isReplacedBy').exists():
+        request.session['validation_error'] = 'An obsoleted resource in the middle of the obsolescence chain cannot be deleted.'
+        return HttpResponseRedirect(request.META['HTTP_REFERER'])
+
+    # when the most recent version of a resource in an obsolescence chain is deleted, the previous
+    # version in the chain needs to be set as the "active" version by deleting "isReplacedBy" relation element
+    if res.metadata.relations.all().filter(type='isVersionOf').exists():
+        is_version_of_res_link = res.metadata.relations.all().filter(type='isVersionOf').first().value
+        idx = is_version_of_res_link.rindex('/')
+        if idx == -1:
+            obsolete_res_id = is_version_of_res_link
+        else:
+            obsolete_res_id = is_version_of_res_link[idx+1:]
+        obsolete_res = get_resource_by_shortkey(obsolete_res_id)
+        if obsolete_res.metadata.relations.all().filter(type='isReplacedBy').exists():
+            eid = obsolete_res.metadata.relations.all().filter(type='isReplacedBy').first().id
+            obsolete_res.metadata.delete_element('relation', eid)
+
     res.delete()
     return HttpResponseRedirect('/my-resources/')
 
+def create_new_version_resource(request, shortkey, *args, **kwargs):
+    res, authorized, user = authorize(request, shortkey, edit=True, full=True, superuser=True)
+
+    try:
+        # lock the resource to prevent concurrent new version creation since only one new version for an obsoleted resource is allowed
+        res.locked = True
+        res.save()
+        # run clear_lock task asynchronously in 2 minutes to ensure lock is cleared when something unexpected happened
+        clear_lock_task = clear_lock.apply_async(args=[shortkey],countdown=120)
```

Comment:
I get what you're going for, but I still think it isn't the best approach.

If you just want to prevent other processes from modifying the same things, should this operation be run in a transaction?